pydantic/_internal/_model_construction.py

```diff
@@ -202,6 +204,11 @@ def build_inner_schema(  # noqa: C901
                 raise
         raise PydanticUndefinedAnnotation(name) from e
 
+    # check for forward references and raise PydanticUndefinedAnnotation
+    for t in type_hints.values():
+        if isinstance(t, typing.ForwardRef):
+            raise PydanticUndefinedAnnotation(t.__forward_arg__)
```

Comment:
Is this necessary? Looks erroneous?

Reply:
See the PR description.

I think this case is tested, @jimkring please confirm.